src/core/mac/sub_mac.cpp

```diff
@@ -328,10 +328,16 @@ void SubMac::ProcessTransmitSecurity(void)
     const ExtAddress *extAddress = nullptr;
     uint8_t           keyIdMode;
 
-    VerifyOrExit(ShouldHandleTransmitSecurity());
     VerifyOrExit(mTransmitFrame.GetSecurityEnabled());
     VerifyOrExit(!mTransmitFrame.IsSecurityProcessed());
 
+    if (!mTransmitFrame.IsARetransmission())
```

Comment:
Probably this check is not needed since Key Id will remain the same across retransmissions anyway.

Reply:
On the other hand why to set it again if it's already set? In any case I'm ok with both options.

Reply:
The only reason not to check is code size. I'm also okay with both options.

Reply:
> Probably this check is not needed since Key Id will remain the same across retransmissions anyway.

Indirect frame retransmission can happen much later and on newly prepared frame (when/if the next data poll is received from the child).